src/urn.cc

```diff
@@ -175,29 +174,17 @@ UrnState::start(HttpRequest * r, StoreEntry * e)
     if (urlres_r == NULL)
         return;
 
-    StoreEntry::getPublic (this, urlres, Http::METHOD_GET);
-}
+    const auto urlEntry = storeGetPublic(urlres, Http::METHOD_GET);
 
-void
-UrnState::fillChecklist(ACLFilledChecklist &checklist) const
-{
-    checklist.setRequest(request.getRaw());
-    checklist.al = ale;
-}
-
-void
-UrnState::created(StoreEntry *e)
-{
-    if (!e || (e->hittingRequiresCollapsing() && !startCollapsingOn(*e, false))) {
+    if (!urlEntry || (urlEntry->hittingRequiresCollapsing() && !startCollapsingOn(*urlEntry, false))) {
         urlres_e = storeCreateEntry(urlres, urlres, RequestFlags(), Http::METHOD_GET);
         sc = storeClientListAdd(urlres_e, this);
         FwdState::Start(Comm::ConnectionPointer(), urlres_e, urlres_r.getRaw(), ale);
-        // TODO: StoreClients must either store/lock or abandon found entries.
-        //if (e)
-        //    e->abandon();
+        if (urlEntry)
+            urlEntry->abandon(__FUNCTION__);
```

Comment:
Let's not leak a pointer to the destroyed urlEntry:

```suggestion
        if (urlEntry) {
            urlEntry->abandon(__FUNCTION__);
            urlEntry = nullptr;
        }
```

We probably should add a `StoreEntry::SafelyAbandon(StoreEntry *&e, const char * context)` static method that will implement it (similar to the existing safe_free() macro), but I do not insist on that because the other abandon() callers touched in this PR currently quit right after they invalidate their StoreEntry pointer.

Reply:
Done.

> I do not insist on that

Left as is.